src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -110,7 +113,11 @@ export default function IndexPatternFactory(Private, Notifier, config, kbnIndex,
 
     if (!indexPattern.fields || !containsFieldCapabilities(indexPattern.fields)) {
       promise = indexPattern.refreshFields();
+    } else if (!indexPattern.metaFieldsVersion) {
+      promise = indexPattern.refreshFields();
+      indexPattern.metaFieldsVersion = true;
```

Comment:
This is related to my comment [earlier in this file](https://github.com/elastic/kibana/pull/8554/files#r98642125). If we make `indexPattern.metaFieldsVersion` a number, then it should probably be initialized to 1 over here, for this initial implementation of meta fields.